thriftbp/admin.go

```diff
@@ -0,0 +1,43 @@
+package thriftbp
+
+import (
+	"net/http"
+
+	"github.com/reddit/baseplate.go/internal/admin"
+	"github.com/reddit/baseplate.go/log"
+)
+
+// AdminServerArgs contain optional configuration for the admin server.
+type AdminServerArgs struct {
+	// AdminAddr is a custom address for the admin server. Defaults to DefaultAdminAddr.
+	AdminAddr string
+}
+
+// NewAdminServer returns a new admin server for internal functionality.
+func NewAdminServer(args *AdminServerArgs) *AdminServer {
+	adminAddr := args.AdminAddr
+	if args.AdminAddr == "" {
+		adminAddr = admin.DefaultAdminAddr
+	}
+
+	return &AdminServer{
+		addr: adminAddr,
+	}
+}
+
+type AdminServer struct {
+	addr string
+}
+
+// Serve starts a blocking HTTP server for internal functions:
+//    metrics       - serve /metrics for prometheus
+//    profiling     - serve /debug/pprof for profiling, ref: https://pkg.go.dev/net/http/pprof
+//
+// Default server address is admin.DefaultAdminAddr.
+//
+// This method is blocking, to prevent blocking run it as a goroutine.
+func (s *AdminServer) Serve() {
+	log.Infof("Serving admin on %s", s.addr)
+	log.Warnw("admin http serve exited", "err", http.ListenAndServe(s.addr, admin.DefaultAdminServeMux))
+	log.Info("admin returnings")
+}
```

Comment:
```suggestion
// ServeAdmin starts a blocking HTTP server for internal functions:
//    metrics       - serve /metrics for prometheus
//    profiling     - serve /debug/pprof for profiling, ref: https://pkg.go.dev/net/http/pprof
//
// Default server address is admin.DefaultAdminAddr.
//
// This function blocks, so it should be run as its own goroutine.
func ServeAdmin() {
	log.Warnw("thriftbp: admin serving exited", "err", admin.Serve())
}
```

Reply:
@kylelemons  I'm fine with making this change, but my one concern is that people won't be able to pass in a different port to run the admin server on. This code will either make people forced to use port 6060 or modify it like this:
```
import "github.com/reddit/baseplate.go/internal/admin"
admin.DefaultAddr = ":8081"
go.ServeAdmin()
```
Modifying it like this seems fine, but if this is how we expect people to modify the port value, then I want to change the name to not say "Default".